### ISSUE TITLE
Is this project still active?

### ISSUE THREAD

**Issue by philgyford** [hMatoba/Piexif#122]
I see there hasn't been an update to the code for a couple of years, there are a bunch of issues with no response, and several outstanding pull requests. Not to blame anyone, because I know everyone has too many things to keep up with :)

I like how Piexif works, and would like to see it continue to be supported, so wanted to check about its likely future...

**Comment by hasii2011** [comment]
+1

**Comment by xOctave** [comment]
+1

**Comment by djjudas21** [comment]
I think the lack of response on this thread gives you the answer :frowning_face: I looked at all the forks too and none of them are active.

It's a shame, because this and [exif](https://gitlab.com/TNThieding/exif) are the only two decent python EXIF modules, and I've hit blockers with both modules, and both seem to be abandoned.

**Comment by darkvertex** [comment]
There's a ton of tags missing in this project. I tried to use it to read some tags and it didn't know about them.

For the most complete I'd recommend checking https://github.com/exiftool/exiftool/
Just look at all the EXIF subformats they can decode:
https://github.com/exiftool/exiftool/tree/master/lib/Image/ExifTool
(Sadly it's in Perl though, not Python, but it's the most complete of all opensource exif projects from what I've seen.)

**Comment by djjudas21** [comment]
@darkvertex exiftool is the gold standard of EXIF modules. I've used it in the past as part of a Perl project, but Perl doesn't get a lot of love these days.

@hMatoba this module is too important to abandon. I totally understand if you're too busy to maintain this, or if you no longer use it yourself, but could you add some other maintainers to this repo so they can do some maintenance, merge some PRs, etc. I'd be happy to volunteer. It seems neater to add collaborators here than to create yet another fork. Thanks

**Comment by djjudas21** [comment]
Judging by @hMatoba's Github profile, there are zero contributions to any project since 2019 and their blog is no longer online. Safe to assume this module is dead.

I'm reluctantly going to fork it and publish under a different name on PyPI... I'm no developer but I can at least read and respond to issues/PRs and keep the module going. I just can't hold up my project any more.

**Comment by davetapley** [comment]
@djjudas21 good timing as I just found this project too.

If you want I'm happy to fork it to [my company's GitHub organization](https://github.com/JEFuller) (and add you as a maintainer) in case you don't want it attached to your personal GitHub account.

Either way instead of re-publishing to PyPi under a different name we should [request a name transfer](https://peps.python.org/pep-0541/#how-to-request-a-name-transfer).
I'm happy to do this as well if easier 💫

**Comment by hasii2011** [comment]
I would be happy to be a co-maintainer on this project

**Comment by djjudas21** [comment]
@davetapley ah great, I didn't know you could do a name transfer. Please go ahead and do that - I'm happy to be a maintainer although I'm not developer. Seems like it will be a better home to have it in a company namespace than a personal one.

**Comment by davetapley** [comment]
Okay forked to https://github.com/JEFuller/Piexif

1. Will use https://github.com/JEFuller/Piexif/projects/1 to track migration
2. @hasii2011 & @djjudas21 I have sent you collaborator requests
3. **Everyone** please 👍🏻 https://github.com/pypa/pypi-support/issues/2065 for visibility

**Comment by djjudas21** [comment]
Thanks @davetapley. I've 👍 your issue with PyPI

**Comment by davetapley** [comment]
Update: 
> will start the reachability process for your request which can take up to 6 weeks. Hopefully we will hear back from the owner soon and proceed with the transfer.

https://github.com/pypa/pypi-support/issues/2065#issuecomment-1194399748

**Comment by davetapley** [comment]
Update: The PyPi project has been transferred to me: https://github.com/pypa/pypi-support/issues/2065#issuecomment-1240500287.

In the coming days I will issue a new release to update ⬇️ and thereafter please use https://github.com/JEFuller/Piexif/issues
https://github.com/hMatoba/Piexif/blob/3422fbe7a12c3ebcc90532d8e1f4e3be32ece80c/setup.py#L23

**Comment by djjudas21** [comment]
Thanks for your perseverance @davetapley. Look forward to a release in its new home :+1:

**Comment by davetapley** [comment]
Making progress, follow along here: https://github.com/JEFuller/Piexif/projects/1

I won't post here again until new release is pushed.

**Comment by Erotemic** [comment]
@davetapley are you still maintaining the library? I get a 404 on the last link you posted.

I've noticed a mistake in the docs, and I was going to fix it, but I'm not sure how active this lib is.

**Comment by davetapley** [comment]
@Erotemic yes, but not actively, I'm not at a computer right now, but I think there was another package which I started using, but I can't remember the name.
Happy to pass along ownership of this, if anyone wants to support it.

**Comment by julianstirling** [comment]
It would be great to be able to start adding typehints into this and keeping the usertags up to date. I am happy to help with or take on maintenance.  I think probably it may be worth creating a piexif organisation, and then we can add maintainers to that?

**Edited to add**

If you are looking at my profile it may look pretty inactive, this is because [I generally work on GitLab](https://gitlab.com/julianstirling). I am in the OpenFlexure microscope core maintenance team, and we use piexif quite heavily so I should be able to find other maintainers.

Also I see you got control of the ReadTheDocs,

**Comment by julianstirling** [comment]
Hi @davetapley , just @-ing you in case you missed the past message.

**Comment by davetapley** [comment]
@julianstirling great! yep your credentials check out and I would love to hand it over to you 🙏🏻 

I think transferring to you would be easier than forking, since you'd keep all the issues / PRs etc. Sound good?


Also I hope not creepy I found you on bsky, I grew up in Weston-super-Mare 😆

**Comment by davetapley** [comment]
> Making progress, follow along here: https://github.com/JEFuller/Piexif/projects/1
> 
> I won't post here again until new release is pushed.

Grr, that's the project where I tracked all the work I did to migrate it from the previous owner, and it's gone 🤯 


Edit:
> Only cards that have been updated in the last year will be migrated.
https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/#github-com-timeline

😒

**Comment by julianstirling** [comment]
> Also I hope not creepy I found you on bsky, I grew up in Weston-super-Mare

Not creepy at all. Glad to keep the project Somerset led!

> I did to migrate it from the previous owner, and it's gone 🤯

That is really not cool! I get them deprecating a feature that isn't useful. But they could have done it in a way where there is a static read-only record of legacy projects. 😠

**Comment by djjudas21** [comment]
Just to chime in, is it worthwhile setting up an organisation to host this project? Then it can have multiple owners in case anyone gets hit by a bus.

I'm a user of Piexif, not really capable of writing serious code, but happy to help with project curation, releases, adding/removing future admins, etc.

I'm also only a few miles up the road in Bristol 😄

**Comment by julianstirling** [comment]
I 100% agree on an organisation. I think having a project in an org, not a personal space signals maturity.

I think this project doesn't really need too much "serious code" right now. Exif data is pretty stable, and the library is pretty functional and importantly lightweight. I think the main thing we need is small bugfixes, good testing, and issue triaging, etc. We also need to make sure that we update stay in-line with modern standards both EXIF as more tags are added, and up to date with Python as it evolves.

_Bristol_!? That isn't even in Somerset! 😆

**Comment by davetapley** [comment]
Sounds good. @julianstirling do you want to create the org?

RIP [Avon](https://en.wikipedia.org/wiki/Avon_(county)) 😑
